src/sqlfluff/core/rules/base.py

```diff
@@ -586,13 +586,37 @@ def _log_critical_errors(error: Exception):  # pragma: no cover
         pass
 
     def _process_lint_result(
-        self, res, templated_file, ignore_mask, new_lerrs, new_fixes
+        self, res, templated_file, ignore_mask, new_lerrs, new_fixes, root
     ):
         self.discard_unsafe_fixes(res, templated_file)
         lerr = res.to_linting_error(rule=self)
         ignored = False
         if lerr:
-            if ignore_mask:
+            # Check whether this should be filtered out for being unparsable.
```

Comment:
It's not clear to me how this code is checking for "unparsable". I see "unparsable" in the comment, but not in the code. How does this situation arise, and how does this code detect and address it? A bit more detailed comments seems like it would be helpful here.

What test(s) are hitting this code? Should we update or add tests for this change?

Do we need to consider `--FIX-EVEN-UNPARSABLE` here? I.e. if that is set, don't filter out the fix?

Reply:
The current configuration for whether rules should work on unparsable sections is in the crawler (see `BaseCrawler`). All of them have the optional attribute of `works_on_unparsable`. That feeds through to `.passes_filter()` (check out the implementation of `BaseCrawler.passes_filter()`). The `passes_filter` method is the one that's called in this section of code - to avoid needing to redefine whether a rule works on unparsable code or not.

The reason I've done it this way is because unless we leverage the existing unparsable logic then we end up redefining it in another place. This way feels more DRY.

Reply:
I hear you on tests though... I'll see what I can do.

Reply:
Definitely agree re: not repeating code. Do you think it'd be worth adding something similar to your PR comment to the code itself, to help future maintainers connect the dots?